config/ingest.go

```diff
@@ -41,3 +50,12 @@ func (c *Ingest) populateUnset() {
 		c.SyncTimeout = def.SyncTimeout
 	}
 }
+
+// EntriesRecursionLimit returns the recursion limit of advertisement entries.
+// See Ingest.EntriesRecursionLimitDepth
+func (c *Ingest) EntriesRecursionLimit() selector.RecursionLimit {
+	if c.EntriesRecursionLimitDepth == 0 {
+		return selector.RecursionLimitNone()
+	}
+	return selector.RecursionLimitDepth(c.EntriesRecursionLimitDepth)
```

Comment:
Another option is to use `0` as a default value, which will get transformed into an actual default by `populateUnset` above, and then use `-1` to mean unlimited.